util/math_helper.h

```diff
@@ -28,6 +28,20 @@ cl::sycl::vec<R, N> run_func_on_vector(funT fun, Args... args) {
   return res;
 }
 
+/* helper for relational functions where true result gives 1 for scalar
+    and -1 for vector argument types */
```

Comment:
Looking at the SYCL 2020 specification, it is not clear.
For example `isunordered` returns a vector of 0 or non null in the  case of `sycl::vec` and a vector of `bool` for `sycl::marray`.
Curious these differences according to the various kinds of functions. So comparing with `== 1` seems OK for `bool` but not for the other case.
So at least just removing ` == 1` looks safer.
But more generally about these differences between the functions, do you think this is a bug in the specification?

Reply:
I don't think it's a bug in the specification - `isunordered` description says it returns non-zero if x or y is NaN and to me it seems to coincide with the idea that these functions return 1 or -1 depending on is it scalar or vector version that referenced for relational functions in 'as defined in the OpenCL 1.2 specification document par. 6.12.6'. May be if it was stated explicitly it would be more clear.

Removing `== 1` would be safer anyway indeed. Done.

Reply:
Thanks.